0.preprocess-sites/8.visualize-plate-layout.py

```diff
@@ -0,0 +1,392 @@
+cp_sat_df = cp_sat_df.drop(["level_3", "cat"], 1)
+cp_sat_df = pd.pivot_table(
+    cp_sat_df, index=["Plate", "Well", "Site", "Ch"], columns=["type"]
+).reset_index()
+cp_sat_df.columns = ["Plate", "Well", "Site", "Ch", "PercentMax", "StdIntensity"]
+
+cp_saturation_gg = (
+    gg.ggplot(cp_sat_df, gg.aes(x="StdIntensity", y="PercentMax", label="Site"))
+    + gg.geom_point()
+    + gg.coord_fixed(ratio=0.25)
+    + gg.geom_text()
+    + gg.facet_wrap(["Ch", "Well"], nrow=len(painting_image_names), scales="free")
+    + gg.theme_bw()
+    + gg.theme(strip_background=gg.element_rect(colour="black", fill="#fdfff4"))
+)
+output_file = pathlib.Path(output_figuresdir, "cp_saturation.png")
+cp_saturation_gg.save(output_file, dpi=300, verbose=False)
+
+# Plots saturation in Barcoding images
+bc_sat_df_cols = ["Plate", "Well", "Site"]
+for x in range(1, (barcoding_cycles + 1)):
+    for nt in nts:
+        bc_sat_df_cols.append(
+            "ImageQuality_PercentMaximal_" + barcoding_prefix + "%02d" % x + "_" + nt
+        )
+        bc_sat_df_cols.append(
+            "ImageQuality_StdIntensity_" + barcoding_prefix + "%02d" % x + "_" + nt
+        )
+bc_sat_df = image_df.loc[:, bc_sat_df_cols]
+
+bc_sat_df = bc_sat_df.set_index(["Plate", "Well", "Site"]).stack().reset_index()
+bc_sat_df[["cat", "type", "Ch"]] = bc_sat_df["level_3"].str.split("_", n=2, expand=True)
+bc_sat_df = bc_sat_df.drop(["level_3", "cat"], 1)
+bc_sat_df = pd.pivot_table(
+    bc_sat_df, index=["Plate", "Well", "Site", "Ch"], columns=["type"]
+).reset_index()
+bc_sat_df.columns = ["Plate", "Well", "Site", "Ch", "PercentMax", "StdIntensity"]
+
+bc_saturation_gg = (
+    gg.ggplot(bc_sat_df, gg.aes(x="StdIntensity", y="PercentMax", label="Site"))
+    + gg.geom_point()
+    + gg.coord_fixed(ratio=0.25)
+    + gg.geom_text()
+    + gg.facet_wrap(["Ch", "Well"], ncol=4, scales="free")
+    + gg.theme_bw()
+    + gg.theme(strip_background=gg.element_rect(colour="black", fill="#fdfff4"))
+)
+output_file = pathlib.Path(output_figuresdir, "bc_saturation.png")
+bc_saturation_gg.save(output_file, dpi=300, verbose=False)
+
+# Create list of questionable channel correlations (alignments)
+corr_df_cols = ["Plate", "Well", "Site", "site"]
+corr_cols = []
+for col in image_df.columns:
+    if "Correlation_Correlation_" in col:
+        corr_cols.append(col)
+        corr_df_cols.append(col)
+image_corr_df = image_df[corr_df_cols]
+image_corr_list = []
+for col in corr_cols:
+    image_corr_list.append(
+        image_corr_df.loc[image_corr_df[col] < correlation_threshold]
+    )
+image_corr_df = pd.concat(image_corr_list).drop_duplicates(subset="site").reset_index()
+for col in corr_cols:
+    image_corr_df.loc[(image_corr_df[col] >= correlation_threshold), col] = "pass"
+
+if len(image_corr_df.index) > 0:
+    corr_output_file = pathlib.Path(results_output, "flagged_correlations.csv")
+    image_corr_df.to_csv(corr_output_file)
```

Comment:
lets leave this in here for now, but this is definitely QC specific right? i.e. not about visualizing plate layout

Reply:
There are a few outputs that aren't plate visualizations (lists of sites with confluent regions, saturation issues, and alignment issues). I wrote them in here as it made sense to me to output the lists along with the visualization of the lists. I suggest that we change the name of the recipe step instead of moving the lists.
How about 8.image-and-segmentation-qc.py?

Reply:
yes, I support that change